packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -107,34 +127,60 @@ module.exports = ({ strapi }) => ({
   async create(body, uid) {
     const modelDef = strapi.getModel(uid);
     const publishData = { ...body };
+    const populateRelations = isRelationsPopulateEnabled(uid);
 
     if (hasDraftAndPublish(modelDef)) {
       publishData[PUBLISHED_AT_ATTRIBUTE] = null;
     }
 
     const params = {
       data: publishData,
-      populate: getDeepPopulate(uid, { countMany: true, countOne: true }),
+      populate: populateRelations ? getDeepPopulate(uid, {}) : getCountDeepPopulate(uid),
     };
 
-    return strapi.entityService.create(uid, params);
+    const entity = await strapi.entityService.create(uid, params);
+
+    // If relations were populated, load the entity again without populating them,
+    // to avoid performance issues
+    if (populateRelations) {
+      return getDeepRelationsCount(entity, uid);
+    }
+
+    return entity;
   },
 
-  update(entity, body, uid) {
+  async update(entity, body, uid) {
     const publishData = omitPublishedAtField(body);
+    const populateRelations = isRelationsPopulateEnabled(uid);
 
     const params = {
       data: publishData,
-      populate: getDeepPopulate(uid, { countMany: true, countOne: true }),
+      populate: populateRelations ? getDeepPopulate(uid, {}) : getCountDeepPopulate(uid),
     };
 
-    return strapi.entityService.update(uid, entity.id, params);
+    const updatedEntity = await strapi.entityService.update(uid, entity.id, params);
+
+    if (populateRelations) {
+      return getDeepRelationsCount(updatedEntity, uid);
```

Comment:
I wonder, what do you think if the function mutate the entity instead of cloning it? I wonder if it would be more efficient, particularly when there are many components.

Reply:
Seems reasonable, haven't thought of merge doing a new copy in each iteration.  I have modified it to use Object.assign instead, wdyt?

Reply:
Yes it's that ! Nice :) I was thinking of going even further, like `setDeepRelationsCount`;

```js
if (populateRelations) {
  setDeepRelationsCount(updatedEntity, uid);
}

return updatedEntity;
```

Reply:
Asynchonous is a bad idea since the entity is used elsewhere, sorry 😬